bazel/MODULE.bazel

```diff
@@ -1 +1,79 @@
-bazel_dep(name = "platforms", version = "0.0.9")
+module(
+    name = "emsdk",
+    version = "4.0.6",
```

Comment:
It looks like this version will need to be updated for each release? 

Does this also mean that only one version of the emsdk module can be used with any given commit in the emsdk repo?  i.e. I can't checkout tip-of-tree and use/install and older version?

Reply:
AFAIK you can use whatever version you want by doing the following in the `MODULE.bazel` file:

```
emscripten_deps = use_extension(
    "@emsdk//:emscripten_deps.bzl",
    "emscripten_deps",
)
emscripten_deps.config(version = "4.0.1")
```

Meaning if you're using module version `4.0.6` you could use up to emsdk version `4.0.6`. 

I just tested using emsdk 4.0.3 with module 4.0.6 and it worked in our main repo.

If you want to create support for a newer version of emsdk or a custom version of it, then you could call `remote_emscripten_repository` directly with `create_toolchains` and `register_toolchains`, and in theory that should also work.

I haven't tested that use case, though.

Reply:
Perhaps we should not tie this version number to the emsdk version number then?  Maybe it should have its own unrelated version.. to avoid confusing these two unrelated versions?

Reply:
I'm torn. I like the idea of versioning schemes being different because they have different lifecycles. At the same time this check in `test_bazel.sh` is an easy way to make sure the module version gets bumped when a new version of emsdk is added: `grep ${VER} bazel/MODULE.bazel || (echo ${FAILMSG} && false)`

IMHO, ideally the versioning schemes would be separated and there would be some kind of check to make sure the module version gets bumped when adding a new emsdk version.

Reply:
I implemented it this way to keep the current behavior, which also has separate versions for emsdk and the emscripten toolchain used.

There is already a check that the bazel module version is updated when adding a new emscripten release. It is the following line in test/test_bazel.sh. :
```
grep ${VER} bazel/MODULE.bazel || (echo ${FAILMSG} && false)
``

Note that scripts/update_bazel_workspace.py also implements automatically updating the module version, so if that script is used to update the repository the version number will also automatically get bumped.

Reply:
> I implemented it this way to keep the current behavior, which also has separate versions for emsdk and the emscripten toolchain used.

But this seems like a change in behaviour:

old: The bazel toolchain version is unrelated to any emsdk version (0.0.9)
new: The bazel toolchain version matches exactly the latest emsdk version (4.0.6)

I'm not saying that new way is wrong, just the it seems like a departure from the old behaviour.

I think I'm find with it as is, but we might also want de-emphasis the ability to install older versions.. since it seems a little odd or use emsdk bazel version 4.0.3 to install emsdk version 4.0.1.  (e.g. the new scheme makes me think they should match more than the old scheme).

Reply:
If the `0.0.9` version you're referring to is the `bazel_dep(name = "platforms", version = "0.0.9")` then that is unrelated to the module version present in

```
module(
    name = "emsdk",
    version = "4.0.6",
)
```

`0.0.9` was the version of the platforms Bazel module that was being used. The source of which can be found here: https://github.com/bazelbuild/platforms

The other one is the version of the module defined by this rule set, i.e., the emsdk Bazel rules.

Reply:
Ah!  I see thanks for the explanation.